src/state.rs

```diff
@@ -835,6 +801,56 @@ impl DemandProxyState {
         Ok(Some(res))
     }
 
+    /// Returns destination address, upstream sans, and final sans, for
+    /// connecting to a remote workload through a gateway.
+    /// Would be nice to return this as an Upstream, but gateways don't necessarily
+    /// have workloads. That is, they could just be IPs without a corresponding workload.
+    pub async fn fetch_network_gateway(
+        &self,
+        gtw: &GatewayAddress,
+        source_workload: &Workload,
+        original_destination_address: SocketAddr,
+        original_us: &Upstream,
+    ) -> Result<(SocketAddr, Vec<Identity>, Vec<Identity>), Error> {
+        match &gtw.destination {
+            Destination::Address(address) => Ok((
+                SocketAddr::from((address.address, gtw.hbone_mtls_port)),
```

Comment:
The intent of these APIs is the address are not raw IPs we send to, but lookup keys into workloads/services. We should use the same logic as fetch_waypoint which calls find_upstream

Reply:
ok, so I can assume that the address has a Workload object behind it